pinot-core/src/main/java/org/apache/pinot/core/operator/filter/predicate/BasePredicateEvaluator.java

```diff
@@ -18,7 +18,25 @@
  */
 package org.apache.pinot.core.operator.filter.predicate;
 
+import org.apache.pinot.common.request.context.predicate.Predicate;
+
+
 public abstract class BasePredicateEvaluator implements PredicateEvaluator {
+  protected final Predicate _predicate;
+
+  protected BasePredicateEvaluator(Predicate predicate) {
+    _predicate = predicate;
+  }
+
+  @Override
+  public Predicate getPredicate() {
+    return _predicate;
+  }
+
+  @Override
+  public Predicate.Type getPredicateType() {
+    return getPredicate().getType();
```

Comment:
Do `_predicate.getType()` instead of function call (which although may not happen at runtime if JVM inlines) ?

Reply:
It should be inlined when there is no other override on `getPredicate()`. It is slightly safer this way in case some class overrides the `getPredicate()`